tests/countries/test_nepal.py

```diff
@@ -0,0 +1,536 @@
+            name, (f"{year}-05-29" for year in (*range(2010, 2019), *range(2021, 2050)))
+        )
+        self.assertNoHolidayName(name, range(2019, 2021))
+
+        # Workdays.
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-05-29" for year in range(2019, 2021))
+        )
+        self.assertNoHolidayName(name, self.workday_holidays, range(2010, 2019), range(2021, 2050))
+
+    def test_constitution_day(self):
+        self.assertHolidayName("Constitution Day", (f"{year}-09-19" for year in range(2010, 2050)))
+
+    def test_christmas(self):
+        self.assertHolidayName("Christmas", (f"{year}-12-25" for year in range(2010, 2050)))
+
+    def test_tamu_losar(self):
+        self.assertHolidayName("Tamu Losar", (f"{year}-12-30" for year in range(2010, 2050)))
+
+    def test_bakrid(self):
+        name = "Bakrid"
+        dt = (
+            "2019-08-12",
+            "2020-08-01",
+            "2021-07-21",
+            "2022-07-10",
+            "2023-06-29",
+            "2024-06-17",
+            "2025-06-07",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2010, 2026))
+
+    def test_bhai_tika(self):
+        name = "Bhai Tika"
+        dt = (
+            "2019-10-29",
+            "2020-11-16",
+            "2021-11-06",
+            "2022-10-26",
+            "2023-11-14",
+            "2024-11-03",
+            "2025-10-23",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2010, 2036))
+
+    def test_buddha_jayanti(self):
+        name = "Buddha Jayanti"
+        dt = (
+            "2019-05-18",
+            "2020-05-07",
+            "2021-05-26",
+            "2022-05-16",
+            "2023-05-05",
+            "2024-05-23",
+            "2025-05-12",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2010, 2036))
+
+    def test_chhath_parwa(self):
+        name = "Chhath Parwa"
+        dt = (
+            "2019-11-02",
+            "2020-11-20",
+            "2021-11-10",
+            "2022-10-30",
+            "2023-11-19",
+            "2024-11-07",
+            "2025-10-28",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2010, 2036))
+
+    def test_duwadashi(self):
+        name = "Duwadashi (Dashain)"
+        dt = (
+            "2021-10-17",
+            "2022-10-07",
+            "2023-10-26",
+            "2024-10-14",
+            "2025-10-04",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, (*range(2010, 2016), *(range(2021, 2036))))
+        self.assertNoHolidayName(name, range(2019, 2021))
+        dt = (
+            "2019-10-10",
+            "2020-10-27",
+        )
+        self.assertHolidayName(name, self.workday_holidays, dt)
+        self.assertNoHolidayName(name, self.workday_holidays, range(2010, 2019), range(2021, 2036))
```

Comment:
```suggestion
        self.assertNoHolidayName(name, self.workday_holidays, range(2010, 2019), range(2021, 2050))
```
And same for all similar holidays.